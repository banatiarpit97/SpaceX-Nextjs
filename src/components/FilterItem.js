import React from 'react';
import styles from './../styles/filter.module.css';

const FilterItem = ({
    type, value, selectedFilters, onFilterSelect
}) => {

    return (

        <span
            onClick={() => onFilterSelect(type, value)}
            className={styles.filter_item}
            style={{ backgroundColor: (selectedFilters[type] == value) ? '#7ab80e' : '#c3de9d' }}
        >
            {value}
        </span>

    )
}

export default FilterItem;