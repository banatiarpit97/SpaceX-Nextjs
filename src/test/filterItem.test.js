import React from 'react';
import renderer from 'react-test-renderer';

import FilterItem from './../components/FilterItem';

it('renders correctly', () => {
  const tree = renderer.create(<FilterItem />).toJSON();
  expect(tree).toMatchSnapshot();
});