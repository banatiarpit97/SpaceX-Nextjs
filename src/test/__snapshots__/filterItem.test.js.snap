// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<span
  className="filter_item"
  onClick={[Function]}
  style={
    Object {
      "backgroundColor": "#c3de9d",
    }
  }
>
  
</span>
`;
